### DISCUSSION TITLE
i.landsat.toar not respecting region?

### DISCUSSION THREAD

**Discussion by CarlosGrohmann** [OSGeo/grass#1853] [Q&A] [answered]
As the title says, I noticed that i.landsat.toar is not respecting the region. 

Is there a reason behind this? Or is it a missing feature? (I'd love to know how to implement it...)

Running GRASS 8.0 dev (26976e33f) on Ubunt 21.04

cheers

**Comment by marisn** [comment]
It seems to be intentional since inception of the module. It should be included in the documentation (open a bug report). @YannChemin or @metzm could explain the reasoning of such approach.
https://github.com/OSGeo/grass/blob/07bd00f3c415d1e8ec258ff94028099d3e716e8d/imagery/i.landsat.toar/main.c#L403

**Reply by veroandreo** [reply]
My guess is that the reason for using the whole raster region instead of the computational region has to do with DOS (dark object substraction); i.e., to have the largest possible sample of pixels from where to get the darkest one of the scene and perform the correction.

In any case, if you don't need the full scene, you could set the region in advance and only import that subset. All of r.in.gdal, r.import and i.landsat.import allow that.

I do agree this behavior should be mentioned in the manual page. Would you mind suggesting a text snippet in a PR, @CarlosGrohmann ?

**Reply by CarlosGrohmann** [reply]
Sure, I'll prepare this

**Reply by CarlosGrohmann** [reply]
To which branch should I submit the PR?

**Reply by veroandreo** [reply]
main, I would say. Then we backport

**Reply by CarlosGrohmann** [reply]
Ok, done

**Comment by YannChemin** [comment, marked as answer]
Dear all, 

Right, the DOS requires the full image, but of course you can create your region of interest, import the band in it and then run the module (rightly said Vero!)

Initially we had the module as a script auto import system, take everything and run pre-processing (i.landsat.toar, i.atcorr, then all the i.akbedo, i.ndvi, i.evapo* etc) on the full tiles, when reaching Level 2 of processing, then see what you have and run your specific model for 10 years of meteo and RS data. So of course, a single image subset area is far from the initial idea. I think there are still some scripts example of that in pygrass wiki for ref.

So yes, unless someone interested in coding the region change, use Vero’s point of view.


Best regards,
Avec mes meilleurs sentiments,

Dr. Yann Chemin
Scientific Officer
European Commission
Joint Research Centre
Directorate D – Sustainable Resources
Via E. Fermi 2749, TP 272
I-21027 Ispra (VA) - Italy
Tel: +39 0332 78 6698
Work (not pgp): ***@***.***
Encrypted (safe): ***@***.***
Encrypted Audio/Video (safe):
jami.net yannchemin

**Reply by CarlosGrohmann** [reply]
Thanks for the answer. I see the reason behind it. I was just a bit confused because it wasn't mentioned in the manual (will write something about it).
